server/github.go

```diff
@@ -102,28 +103,34 @@ func NewReleaseManager(owner string, repo string) *ReleaseManager {
 		latestAssetsMap: make(map[string]map[string]*Asset),
 	}
 
+	if mockServerAddr != "" {
+		uri, err := url.Parse("http://" + mockServerAddr)
+		if err != nil {
+			panic(err.Error())
+		}
+		ghc.client.BaseURL = uri
+		ghc.client.UploadURL = uri
+		log.Debugf("Mocking Github API.")
+	}
+
 	return ghc
 }
 
 // getReleases queries github for all product releases.
 func (g *ReleaseManager) getReleases() ([]Release, error) {
-	var releases []Release
+	releases := []Release{}
 
 	for page := 1; true; page++ {
 		opt := &github.ListOptions{Page: page}
 
 		rels, _, err := g.client.Repositories.ListReleases(g.owner, g.repo, opt)
-
 		if err != nil {
 			return nil, err
 		}
-
 		if len(rels) == 0 {
 			break
 		}
 
-		releases = make([]Release, 0, len(rels))
```

Comment:
This thing was always overwriting the list of releases with whatever was on top of the listing, this was good because that was usually the latest release, until something else made it to the top.